File: trikot-viewmodels/sample/android/src/main/java/com/trikot/viewmodels/sample/resource/SampleImageResourceProvider.kt
package com.trikot.viewmodels.sample.resource

import android.content.Context
import com.mirego.trikot.viewmodels.resource.TrikotImageResource
import com.mirego.trikot.viewmodels.resources.ImageViewModelResourceProvider
import com.trikot.viewmodels.sample.R

class SampleImageResourceProvider : ImageViewModelResourceProvider {
    override fun resourceIdFromResource(resource: TrikotImageResource, context: Context): Int? {
        if (resource == TrikotImageResources.ICON) {
            return R.drawable.icon
        }
        return null
    }
}

File: trikot-viewmodels/sample/ios/iosApp/sample/SampleImageResourceProvider.swift
import Foundation
import SampleTrikotFrameworkName
import Trikot

class SampleImageResourceProvider: ImageViewModelResourceProvider {
    func image(fromResource resource: TrikotImageResource?) -> UIImage? {
        if (resource as? TrikotImageResources) == TrikotImageResources.icon {
            return #imageLiteral(resourceName: "icon")
        }
        return nil
    }
}
